### PR TITLE
add enumerable to dolly collection completes #87

### PR DESCRIPTION
**Pull request by seancookr**
https://github.com/amco/dolly/issues/87

This adds the entire [Enumerable](http://ruby-doc.org/core-2.3.1/Enumerable.html) module to the Dolly Collection, all defined from the `each` method.  

I have neglected to add tests because the current collection tests should be proof of the validity of this fix https://github.com/amco/dolly/blob/87-add-enumerable-to-dolly-collection/test/collection_test.rb

EDIT:

Second attempt, use the class as a delegate of Set so that the modules are included naturally.

### CODE REVIEW

File: lib/dolly/collection.rb [hunk 1]
```diff
@@ -12,10 +13,6 @@ def initialize str, docs_class
       @json = str
     end
 
-    def first
-      to_a.first
-    end
-
     def last
       to_a.last
```
Comment (seancookr) [inline lib/dolly/collection.rb]: last is the only method not defined on enumerable

File: lib/dolly/collection.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 module Dolly
   class Collection
     extend Forwardable
+    include Enumerable
```
Comment (netlogics) [inline lib/dolly/collection.rb]: 👏

**Comment by netlogics** [comment]
:shipit:

File: lib/dolly/collection.rb (hunk 2 above)
Comment (rubenrails) [inline lib/dolly/collection.rb]: fuckyeah

**Comment by rubenrails** [comment]
:shipit:

File: lib/dolly/collection.rb (hunk 2 above)
Comment (javierg) [inline lib/dolly/collection.rb]: including enumerable will remove the need to use forwardable and forward the methods like `+` or `-` to the collection object.... this is a different approach... originally all the enum methods where kept away on purpose.... so more refactor is required if we will do this

Comment (seancookr) [inline lib/dolly/collection.rb]: @javierg I am under the impression that both can be in the class and live in harmony.

File: lib/dolly/collection.rb [hunk 3]
```diff
@@ -1,19 +1,14 @@
...
-      @set = Set.new
       @docs_class = docs_class
       @json = str
-    end
-
-    def first
-      to_a.first
+      initial = []
+      super(initial)
+      load
```
Comment (javierg) [inline lib/dolly/collection.rb]: dont you need to do the load before calling super?

Comment (seancookr) [inline lib/dolly/collection.rb]: no actually. that was what my problem was doing this for the last 2 hours. Without doing super, the set is never initialized, and during the `load` method at some point `self << document` is called, which cant happen if it isn't initialized

Comment (javierg) [inline lib/dolly/collection.rb]: right

**Comment by javierg** [comment]
:shipit:

**Pull request merged**